File: xdm-app/src/main/java/xdm/app/service/AppControllerService.java
package xdm.app.service;

import xdm.app.data.DbRecord;
import xdm.app.models.BrowserDownloadInfo;

public interface AppControllerService {
  void run(String[] args);

  void showAppWindow();

  void hideDownloadProgressWindow(long id);

  void showDownloadProgressWindow(long id);

  void showDownloadCompleteWindow(long id, String folder, String fileName);

  void updateDownloadInView(long id);

  void updateDownloadInView(DbRecord entry);

  void addDownloadInView(long id);

  void addDownloadInView(DbRecord entry);

  //  void updateProgressWindow(long id, AbstractDownloader downloader);

  void showErrorInProgressWindow(long id, String errorMessage);

  void addDownload(final BrowserDownloadInfo metadata);

  void addVideoDownload(long vid, String fileName, long fileSize, String fileType);
}

File: xdm-app/src/main/java/xdm/app/service/AppControllerService.kt
package xdm.app.service

import xdm.core.downloaders.HttpDownloadTaskInfo

interface AppControllerService {
    fun run(args: Array<String>)

    fun showAppWindow()

    fun hideDownloadProgressWindow(id: Long)

    fun showDownloadProgressWindow(id: Long)

    fun showDownloadCompleteWindow(id: Long, folder: String, fileName: String)

    fun updateDownloadInView(id: Long)

    fun addDownloadInView(id: Long)

    //  void updateProgressWindow(long id, AbstractDownloader downloader);
    fun showErrorInProgressWindow(id: Long, errorMessage: String)

    fun addDownload(metadata: HttpDownloadTaskInfo?)

    fun addVideoDownload(vid: Long, fileName: String, fileSize: Long?, fileType: String?)
}
